bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/BigtableHBaseVeneerSettings.java

```diff
@@ -696,17 +650,18 @@ private void configureNonIdempotentCallSettings(UnaryCallSettings.Builder unaryC
 
       StatusCode.Code code = StatusCode.Code.valueOf(trimmed);
       Preconditions.checkNotNull(code, String.format("Unknown status code %s found", stringCode));
-      statusCodeBuilder.add(code);
-      LOG.debug("gRPC retry on: %s", stringCode);
+      codes.add(code);
     }
-    return statusCodeBuilder.build();
-  }
 
-  private boolean isBatchModeEnabled() {
-    return configuration.getBoolean(BIGTABLE_USE_BATCH, false);
-  }
+    String enableDeadlineRetry = configuration.get(ENABLE_GRPC_RETRY_DEADLINEEXCEEDED_KEY);
+    if (!Strings.isNullOrEmpty(enableDeadlineRetry)) {
+      if (Boolean.parseBoolean(enableDeadlineRetry)) {
+        codes.add(StatusCode.Code.DEADLINE_EXCEEDED);
+      } else {
+        codes.remove(StatusCode.Code.DEADLINE_EXCEEDED);
```

Comment:
if DEADLINE_EXCEEDED already there, do we need to add it? (in the if statement)

Reply:
not sure what you mean? Can you rephrase?

Reply:
I didnt want to make assumptions about veneer defaults here.